### DISCUSSION TITLE
best way to enable dns fowarding to local dns

### DISCUSSION THREAD

**Discussion by davidmacdonald80** [k3s-io/k3s#10341] [General]
I'm still pretty new to Kubernetes/k3s.  I'm not sure how many pages I've skimmed through trying to find the right, best, or any way to forward  to a local dns server outside of Kubernetes.  I've figured out how to modify the deployment configmap but it doesn't seem like a permanent or best solution. I'm fine with a permanent patch or reinstalling  from scratch with the proper flags or configs.  I just have no clue what they are.

**Comment by brandond** [comment]
What specifically are you trying to forward? Coredns will use whatever nameservers you have in resolv.conf on the node where the pod runs, so you should just be able to point the node at your preferred DNS server, and anything it can resolve, can be resolved from within the cluster.

**Reply by davidmacdonald80** [reply]
Thanks,  The way you said that makes me feel silly for not figuring it out on my own.   Pointing to the hosts resolv.conf was exactly what I was wanting.

**Reply by SpootyMcSpoot** [reply]
I find this strange as my instance is configure to read from /etc/resolv.conf and yet the /etc/resolv.conf within pods only points to the internal nameserver. Therefore I am unable to resolve say google.com.

**Reply by davidmacdonald80** [reply]
What is currently in your resolv.conf on the host?  I had to update /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf so that the resolv.conf didn't use 127.0.0.1.  Depending on your system, there are a couple other ways to keep it from using the localhost instead of the actual DNS IPs.
